src/data/reports/glass-data-submission/GLASSDataSubmissionDefaultRepository.ts

```diff
@@ -725,63 +745,116 @@ export class GLASSDataSubmissionDefaultRepository implements GLASSDataSubmission
+                );
+                break;
             }
-            if (module === "AMR - Individual") {
-                const amrIndividualPrograms =
-                    modules.find(module => module.name === "AMR - Individual")?.programs ?? [];
+            case AMR_INDIVIDUAL: {
+                const amrIndividualQuestionnaires =
+                    modules.find(module => module.name === AMR_INDIVIDUAL)?.questionnaires ?? [];
+                const amrIndividualPrograms = modules.find(module => module.name === AMR_INDIVIDUAL)?.programs ?? [];
+
+                _.forEach(amrIndividualQuestionnaires, async amrIndividualQuestionnaire => {
+                    await this.duplicateDataSet(amrIndividualQuestionnaire, items);
+                });
                 _.forEach(amrIndividualPrograms, async amrIndividualProgram => {
-                    await this.duplicateProgramStages(amrIndividualProgram, items);
+                    await this.duplicateTrackerProgram(amrIndividualProgram, items);
                 });
+                break;
             }
-            if (module === "EGASP") {
-                const egaspPrograms = modules.find(module => module.name === "EGASP")?.programs ?? [];
-                _.forEach(egaspPrograms, async egaspProgram => await this.duplicateProgram(egaspProgram, items));
+            case EGASP: {
+                const egaspPrograms = modules.find(module => module.name === EGASP)?.programs ?? [];
+                _.forEach(
+                    egaspPrograms,
+                    async egaspProgram =>
+                        await this.duplicateEventProgram(egaspProgram, items, selectedModule === EGASP)
+                );
+                break;
             }
+        }
+    }
 
-            const newSubmissionValues = this.getNewSubmissionValues(items, objects, "APPROVED");
-            const recipients = await this.getRecipientUsers(items, modules);
-
-            const message = await this.getNotificationText(items, modules, "approved");
-            this.sendNotifications(message, message, [], recipients);
-
-            return await this.globalStorageClient.saveObject<GLASSDataSubmissionItem[]>(namespace, newSubmissionValues);
-        } else if (signals) {
-            const objects = await this.globalStorageClient.listObjectsInCollection<EARDataSubmissionItem>(namespace);
-            const newSubmissionValues = this.getNewEARSubmissionValues(signals, objects, "APPROVED");
-            const recipients = await this.getEARRecipientUsers(signals, modules);
-
-            const message = this.getEARNotificationText(signals, modules, "approved");
-            this.sendNotifications(message, message, [], recipients);
+    private async getAMCProductRegisterId() {
+        const { objects } = await this.api.models.programs
+            .get({
+                fields: { id: true },
+                filter: { name: { eq: "AMC - Product Register" } },
```

Comment:
better to filter by program id or program code than name as the the name could change. Also add the id as a constant and use whenever required.

Reply:
i will filter by code. is it a good idea to use the id as a constant as opposed to getting the id from the api (using the code or name)? is the id always the same?

Reply:
yes, better to use code. Id could be different in different env (prod, dev...)